### PR TITLE
Fix spline color term apply code so that it returns the target flux.

### PR DESCRIPTION
**Pull request by erykoff**

### CODE REVIEW

File: python/lsst/the/monster/splinecolorterms.py [hunk 1]
```diff
@@ -333,11 +333,11 @@ def apply(self, source_color_flux_1, source_color_flux_2, source_flux):
 
         mag_color = mag_1 - mag_2
 
-        model = self.spline.interpolate(mag_color)
-        model -= self.flux_offset/source_flux
+        target = self.spline.interpolate(mag_color) * source_flux
```
Comment (psferguson) [inline python/lsst/the/monster/splinecolorterms.py:336]: are you sure target = model * source_flux 
based off the cost function https://github.com/lsst-dm/the_monster/blob/08b3df7e1f1a56311da2aa0d2f48d0b714a39770/python/lsst/the/monster/splinecolorterms.py#L206

I would have assumed target_flux = source_flux / model

**Review by psferguson — CHANGES_REQUESTED** [review]
I also think you will need to update the qa plots to deal with the new format of the apply output

I think this is the only time that happens: https://github.com/lsst-dm/the_monster/blob/1cbe00db062e69233c546841f3a7c03bcfa7b3db/python/lsst/the/monster/measure_colorterms.py#L198

**Comment by erykoff** [comment]
You're right, this was backwards and mixed up.  I messed up the application, as well as the flux offset.  This has been fixed, but I need to remeasure the color terms.  Those will be pushed on this PR in a few minutes.

**Review by psferguson — APPROVED** [review]
This looks good, much easier for me to follow along the details of applying the model.

And the unit test looks more robust

**Pull request merged**